### ISSUE TITLE
Redirecting stdout+stderr using `pnpm install &> output.log` wraps lines

### ISSUE THREAD

**Issue by vjpr** [pnpm/pnpm#873]
```
WARN Refetching /Users/Vaughan/.pnpm-store/2/registry.npmjs.org/zen-observable-t
s/0.4.0 to store, as it was modified
Resolving: total 2350, reused 2339, downloaded 11, done
Creating dependency tree
WARN registry.npmjs.org/deck.gl/4.1.0 requires a peer of luma.gl@>=4.0.0-beta.5
but none was installed.
WARN registry.npmjs.org/fixed-data-table/0.6.0 requires a peer of react@>=0.13.0
 <0.15.0 || ^0.14.0-beta3 but version 15.6.1 was installed.
WARN registry.npmjs.org/fixed-data-table/0.6.0 requires a peer of react-dom@>=0.
14.0 <0.15.0 || ^0.14.0-beta3 but version 15.6.1 was installed.
WARN registry.npmjs.org/graphql-bookshelf/0.0.8 requires a peer of graphql@^0.7.
0 but version 0.10.5 was installed.
WARN registry.npmjs.org/graphql-relay/0.4.4 requires a peer of graphql@^0.5.0 ||
 ^0.6.0 || ^0.7.0 || ^0.8.0-b but version 0.10.5 was installed.
WARN registry.npmjs.org/graphql-thinky/0.4.0-rc-3 requires a peer of graphql@^0.
8.0 but version 0.10.5 was installed.
```

### Additional information:

 - `node -v` prints: 8.1.2
 - Windows, OS X, or Linux?: macOS + Linux

**Comment by vjpr** [comment]
Looks like the issue in is `pnpm-default-reporter > log-update`, mentioned here: https://github.com/sindresorhus/log-update/pull/21.

**Comment by zkochan** [comment]
Related: https://github.com/pnpm/pnpm/issues/572#issuecomment-297551737

**Comment by vjpr** [comment]
Instead of using `log-update` to do non-fixed writes, could we not just use `console.log`?

**Comment by zkochan** [comment]
Probably. Here's the code that should be changed: https://github.com/pnpm/pnpm-default-reporter/blob/master/src/terminalWriter.ts

**Comment by zkochan** [comment]
> Instead of using log-update to do non-fixed writes, could we not just use console.log?

this part will be fixed by https://github.com/sindresorhus/log-update/pull/27

**Comment by zkochan** [comment]
Doing an append-only reporter today. It will be used on CI servers by default. Currently, pnpm is silent by default on CI servers.

So it will be possible to use something like `pnpm install --reporter append-only` and the output will be only appended with `console.log`. 

Maybe someone knows how to feature-detect when it is possible to rewrite the output. In that case, we can prefer this reporter not only on CI servers.

**Comment by etamponi** [comment]
I'd say you can check if stdout is not a tty, in that case you want an
append-only log.
https://stackoverflow.com/questions/20585729/how-to-detect-if-nodes-process-stdout-is-being-piped

Il 14 gen 2018 10:43, "Zoltan Kochan" <notifications@github.com> ha scritto:

**Comment by zkochan** [comment]
thanks @etamponi, I used your suggestion

@vjpr this is fixed in `pnpm@1.29.1`
